tests/test_algorithm.py

```diff
@@ -3531,6 +3532,42 @@ def handle_data(algo, data):
                                        env=self.env)
         self.check_algo_succeeds(algo, handle_data)
 
+    def test_set_min_leverage(self):
+        # Algo date is not past deadline so algorithm succeeds
+        offset = pd.Timedelta('10 days')
+
+        # Set min leverage to 0 so buying one share fails.
+        def handle_data(algo, data):
+            algo.order(algo.sid(self.sidint), 1)
+            algo.record(latest_time=algo.get_datetime())
+
+        algo = SetMinLeverageAlgorithm(1, offset, sim_params=self.sim_params,
+                                       env=self.env)
+        self.check_algo_succeeds(algo, handle_data)
+
+        # Algo date is past deadline
+        # Set min leverage to 1 so buying one share fails.
+        offset = pd.Timedelta('1 day')
+
+        def handle_data(algo, data):
+            algo.order(algo.sid(self.sidint), 1)
+            algo.record(latest_time=algo.get_datetime())
+
+        algo = SetMinLeverageAlgorithm(1, offset, sim_params=self.sim_params,
+                                       env=self.env)
+        self.check_algo_fails(algo, handle_data)
+        self.assertEqual(
+            algo.recorded_vars['latest_time'],
+            pd.Timestamp('2006-01-04 21:00:00', tz='UTC'),
```

Comment:
Can we also test that this fails at 01-05 if we increase the offset delta to 2 days?  (I might just break out those two cases into separate test methods instead of having all the cases at once).